src/H5RT.c

```diff
@@ -0,0 +1,700 @@
+    assert(num_slabs <= node_capacity);
+done:
+    if (ret_value == SUCCEED) {
+        *slab_count_out = num_slabs;
+        *slab_size_out  = slab_size;
+    }
+    FUNC_LEAVE_NOAPI(ret_value)
+}
+
+/* Load the provided leaves into the r-tree in an efficient manner.
+ *
+ * Parameters:
+ *   node = the node to fill
+ *   rank = the rank of the hyper-rectangles
+ *   leaves = a pointer to the first leaf in this block
+ *   count = the number of leaves in this block
+ *   root = whether this is the root node
+ *   prev_sort_dim = the dimension that was last sorted on (or -1 if none)
+ * This is an implementation of the sort-tile-recursive (STR) algorithm.
+ * See "STR: A Simple and Efficient Algorithm for R-Tree Packing"
+ * https://archive.org/details/nasa_techdoc_19970016975/page/n9 */
+static herr_t
+H5RT__bulk_load(H5RT_node_t *node, int rank, H5RT_leaf_t *leaves, size_t count, int prev_sort_dim)
+{
+    herr_t       ret_value        = SUCCEED;
+    size_t       leaves_left      = 0; /* Leaves left to partition */
+    size_t       child_leaf_count = 0;
+    H5RT_leaf_t *child_leaf_start = NULL;
+
+    bool this_rank_sorted = false;
+    int  sort_dim         = -1;
+
+    size_t num_slabs = 0;
+    size_t slab_size = 0;
+
+    FUNC_ENTER_PACKAGE
+
+    if (!node)
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "invalid node");
+    if (rank < 1 || rank > H5S_MAX_RANK)
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "invalid rank");
+    if (!leaves)
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "invalid leaves");
+    if (count == 0)
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "must have at least one leaf");
+    if (prev_sort_dim < -1)
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "invalid previous sort dimension");
+
+    this_rank_sorted = (prev_sort_dim == (rank - 1));
+
+    /* Compute the max/min bounds of the provided node */
+    /* Initial values */
+    for (int i = 0; i < rank; i++) {
+        node->min[i] = leaves[0].min[i];
+        node->max[i] = leaves[0].max[i];
+    }
+    /* Compute max/min from leaves */
+    for (size_t i = 0; i < count; i++) {
+        for (int d = 0; d < rank; d++) {
+            if (leaves[i].min[d] < node->min[d])
+                node->min[d] = leaves[i].min[d];
+            if (leaves[i].max[d] > node->max[d])
+                node->max[d] = leaves[i].max[d];
+        }
+    }
+
+    if (count <= H5RT_MAX_NODE_SIZE) {
+        /* Base Case - All leaves will fit into this node */
+        node->nchildren           = (int)count;
+        node->children_are_leaves = true;
+        node->children.leaves     = leaves;
+    }
+    else {
+        /* Recursive case - there will be child nodes */
+        node->children_are_leaves = false;
+
+        /* Sort hyper-rectangles in this region by the first unsorted coordinate of their midpoints */
+        if (!this_rank_sorted) {
+            assert(prev_sort_dim < rank - 1);
+            sort_dim = prev_sort_dim + 1;
```

Comment:
So it just sorts dimensions in the order 0,1,2, 3,3,3,3 (for rank 3)?

Reply:
Would it be hard to use something simple like picking the largest dimension? I'm wondering how much this would help performance